### ISSUE TITLE
Investigate failing `test_hbavss_light_parallel_share_array_open` test

### ISSUE THREAD

**Issue by Drake-Eidukas** [initc3/HoneyBadgerMPC#284]
Currently, the batch_reconstruction function will fail to reconstruct the values in this unit test about half the time. Here's the stacktrace:
```
tests/test_hbavss.py:649: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
honeybadgermpc/mpc.py:280: in join
    return await asyncio.gather(*self.tasks)
honeybadgermpc/mpc.py:216: in _run
    return result.result()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

context = <honeybadgermpc.mpc.Mpc object at 0x7fe1e8c69da0>

    async def _prog(context):
        share_values = list(map(context.field, shares[context.myid]))
>       opened_shares = set(await context.ShareArray(share_values).open())
E       honeybadgermpc.exceptions.HoneyBadgerMPCError: Batch reconstruction failed!

tests/test_hbavss.py:638: HoneyBadgerMPCError
-------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------
2019-05-02 17:22:04,689:[reed_solomon.py:297]:[CRITICAL]: Optimistic decoding failed
2019-05-02 17:22:04,691:[batch_reconstruction.py:177]:[ERROR]: [BatchReconstruct] P1 reconstruction failed!
2019-05-02 17:22:04,693:[reed_solomon.py:297]:[CRITICAL]: Optimistic decoding failed
2019-05-02 17:22:04,694:[batch_reconstruction.py:177]:[ERROR]: [BatchReconstruct] P1 reconstruction failed!
2019-05-02 17:22:04,695:[reed_solomon.py:297]:[CRITICAL]: Optimistic decoding failed
2019-05-02 17:22:04,695:[batch_reconstruction.py:177]:[ERROR]: [BatchReconstruct] P1 reconstruction failed!
2019-05-02 17:22:04,696:[reed_solomon.py:297]:[CRITICAL]: Optimistic decoding failed
2019-05-02 17:22:04,696:[batch_reconstruction.py:177]:[ERROR]: [BatchReconstruct] P1 reconstruction failed!
2019-05-02 17:22:04,697:[reed_solomon.py:297]:[CRITICAL]: Optimistic decoding failed
2019-05-02 17:22:04,697:[batch_reconstruction.py:177]:[ERROR]: [BatchReconstruct] P1 reconstruction failed!
2019-05-02 17:22:04,698:[reed_solomon.py:297]:[CRITICAL]: Optimistic decoding failed
2019-05-02 17:22:04,698:[batch_reconstruction.py:177]:[ERROR]: [BatchReconstruct] P1 reconstruction failed!
2019-05-02 17:22:04,699:[mpc.py:164]:[ERROR]: Batch reconstruction for share_array (id: 0) failed!
2019-05-02 17:22:04,699:[mpc.py:164]:[ERROR]: Batch reconstruction for share_array (id: 0) failed!
2019-05-02 17:22:04,699:[mpc.py:164]:[ERROR]: Batch reconstruction for share_array (id: 0) failed!
2019-05-02 17:22:04,699:[mpc.py:164]:[ERROR]: Batch reconstruction for share_array (id: 0) failed!
2019-05-02 17:22:04,699:[mpc.py:164]:[ERROR]: Batch reconstruction for share_array (id: 0) failed!
2019-05-02 17:22:04,700:[mpc.py:164]:[ERROR]: Batch reconstruction for share_array (id: 0) failed!
---------------------------------------------------------------------- Captured log call ----------------------------------------------------------------------
reed_solomon.py            297 CRITICAL Optimistic decoding failed
batch_reconstruction.py    177 ERROR    [BatchReconstruct] P1 reconstruction failed!
reed_solomon.py            297 CRITICAL Optimistic decoding failed
batch_reconstruction.py    177 ERROR    [BatchReconstruct] P1 reconstruction failed!
reed_solomon.py            297 CRITICAL Optimistic decoding failed
batch_reconstruction.py    177 ERROR    [BatchReconstruct] P1 reconstruction failed!
reed_solomon.py            297 CRITICAL Optimistic decoding failed
batch_reconstruction.py    177 ERROR    [BatchReconstruct] P1 reconstruction failed!
reed_solomon.py            297 CRITICAL Optimistic decoding failed
batch_reconstruction.py    177 ERROR    [BatchReconstruct] P1 reconstruction failed!
reed_solomon.py            297 CRITICAL Optimistic decoding failed
batch_reconstruction.py    177 ERROR    [BatchReconstruct] P1 reconstruction failed!
mpc.py                     164 ERROR    Batch reconstruction for share_array (id: 0) failed!
mpc.py                     164 ERROR    Batch reconstruction for share_array (id: 0) failed!
mpc.py                     164 ERROR    Batch reconstruction for share_array (id: 0) failed!
mpc.py                     164 ERROR    Batch reconstruction for share_array (id: 0) failed!
mpc.py                     164 ERROR    Batch reconstruction for share_array (id: 0) failed!
mpc.py                     164 ERROR    Batch reconstruction for share_array (id: 0) failed!
```

**Comment by Drake-Eidukas** [comment]
This used to simply cause `batch_reconstruction` to return `None` to opening a share array, now it raises an exception, however, this test failed even before the change to raise an exception.

**Comment by Drake-Eidukas** [comment]
In order to induce this failure, I could run using `docker-compose --rm ... pytest --dist=loadfile -nauto`, or `... pytest --pdb`

**Comment by Drake-Eidukas** [comment]
Fixed by #287
